tests/e2e/e2e_garbage_collector_test.go

```diff
@@ -0,0 +1,135 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openservicemesh/osm/pkg/constants"
+	. "github.com/openservicemesh/osm/tests/framework"
+)
+
+var _ = OSMDescribe("Test garbage collection for unused envoy bootstrap config secrets",
+	OSMDescribeInfo{
+		Tier:   2,
+		Bucket: 2,
+	},
+	func() {
+		Context("Garbage Collection", func() {
+			userAppBaseName := "app"
+
+			numberOfUserServices := 2
+			userReplicaSet := 5
+
+			var userServices []string
+
+			for i := 0; i < numberOfUserServices; i++ {
+				userServices = append(userServices, fmt.Sprintf("%s%d", userAppBaseName, i))
+			}
+
+			// Used across the test to wait for concurrent steps to finish
+			var wg sync.WaitGroup
+
+			It("Tests garbage collections", func() {
+				// Install OSM
+				Expect(Td.InstallOSM(Td.GetOSMInstallOpts())).To(Succeed())
+
+				// Create NSs
+				Expect(Td.CreateMultipleNs(userServices...)).To(Succeed())
+				Expect(Td.AddNsToMesh(true, userServices...)).To(Succeed())
+
+				// User apps
+				for _, userApp := range userServices {
+					svcAccDef, deploymentDef, svcDef := Td.SimpleDeploymentApp(
+						SimpleDeploymentAppDef{
+							Name:         userApp,
+							Namespace:    userApp,
+							ReplicaCount: int32(userReplicaSet),
+							Command:      []string{"/bin/bash", "-c", "--"},
+							Args:         []string{"while true; do sleep 30; done;"},
+							Image:        "songrgg/alpine-debug",
+							Ports:        []int{80},
+						})
+
+					_, err := Td.CreateServiceAccount(userApp, &svcAccDef)
+					Expect(err).NotTo(HaveOccurred())
+					_, err = Td.CreateDeployment(userApp, deploymentDef)
+					Expect(err).NotTo(HaveOccurred())
+					_, err = Td.CreateService(userApp, svcDef)
+					Expect(err).NotTo(HaveOccurred())
+
+					wg.Add(1)
+					go func(app string) {
+						defer GinkgoRecover()
+						defer wg.Done()
+						Expect(Td.WaitForPodsRunningReady(app, 200*time.Second, userReplicaSet)).To(Succeed())
+					}(userApp)
+				}
+
+				wg.Wait()
+
+				By("Verifying the secrets have been patched with OwnerReference")
+
+				podSelector := constants.EnvoyUniqueIDLabelName
+				for _, ns := range userServices {
+					pods, err := Td.Client.CoreV1().Pods(ns).List(context.Background(), metav1.ListOptions{LabelSelector: podSelector})
+					Expect(err).To(BeNil())
+
+					for _, pod := range pods.Items {
```

Comment:
reducing the replica count would eliminate this loop

Reply:
Removed